pkg/snet/network.go

```diff
@@ -146,6 +146,12 @@ func New(conf Config, eb *appevent.Broadcaster) (*Network, error) {
 			SK:        conf.SecKey,
 			Table:     pktable.NewTable(conf.NetworkConfigs.STCP.PKTable),
 			LocalAddr: conf.NetworkConfigs.STCP.LocalAddr,
+			BeforeDialCallback: func(network, addr string) error {
+				data := appevent.TCPDialData{RemoteNet: network, RemoteAddr: addr}
+				event := appevent.NewEvent(appevent.TCPDial, data)
+				_ = eb.Broadcast(context.Background(), event) //nolint:errcheck
```

Comment:
Could you please explain why an error is ignored here instead of returning?

Reply:
@nkryuchkov dial will be cancelled in case of error, cause callback is fired before dialing